### PR TITLE
add JA4 test cases -- v6

### PR DESCRIPTION
**Pull request by satta**
Previous PR: #1564 

This PR introduces test cases for TLS and QUIC JA4 calculation.
Requires Suricata PR https://github.com/OISF/suricata/pull/10095

Changes to previous PR:
* Rebase against current master to match up expectations with current Suricata master behaviour.
* Add "requires: feature" tests for regular cases as well as for cases where JA3/JA4 are not compiled in or disabled.

## Ticket

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/6379

### CODE REVIEW

File: tests/ja4-quic/test.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+requires:
+  min-version: 7.0.0
```
Comment (catenacyber) [inline tests/ja4-quic/test.yaml:2]: Should be 8 ?

File: tests/ja4-rules-invalid/test.rules [hunk 2]
```diff
@@ -0,0 +1 @@
+alert smb any any -> any any (msg:"JA4 Broken Test 1"; ja4.hash; content: "q"; sid:1;)
```
Comment (catenacyber) [inline tests/ja4-rules-invalid/test.rules:1]: It is broken because of `smb` right ?

File: tests/ja4-rules-requires-off/test.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+requires:
+  min-version: 7.0.3
+  files:
+    - rust/src/ja4.rs
+  script:
+    - grep 'ProvidesFeature(FEATURE_JA4);' src/app-layer-ssl.c > /dev/null
```
Comment (catenacyber) [inline tests/ja4-rules-requires-off/test.yaml:6]: I do not think this grep is needed, if you have the next one

File: tests/ja4-rules-requires-off/test.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    - ./src/suricata --build-info | grep "JA4 support" | grep no > /dev/null
```
Comment (catenacyber) [inline tests/ja4-rules-requires-off/test.yaml:7]: Why not use `--dump-features` ?

File: tests/ja4-rules-requires/test.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+requires:
+  min-version: 7.0.3
+  files:
+    - rust/src/ja4.rs
+  script:
+    - grep 'ProvidesFeature(FEATURE_JA4);' src/app-layer-ssl.c > /dev/null
```
Comment (catenacyber) [inline tests/ja4-rules-requires/test.yaml:6]: Why not use --dump-features here ? What is the difference with the previous test ?

File: tests/tls/tls-ja3s/test.yaml [hunk 6]
```diff
@@ -6,6 +6,7 @@ requires:
     - HAVE_LIBJANSSON
     - HAVE_NSS
     - HAVE_LUA
+    - HAVE_JA3
```
Comment (catenacyber) [inline tests/tls/tls-ja3s/test.yaml:9]: Could be its own prerequisite commit

File: tests/ja4-rules-invalid/test.rules (hunk 2 above)
Comment (satta) [reply tests/ja4-rules-invalid/test.rules:1]: Right, wrong protocol.

File: tests/ja4-quic/test.yaml (hunk 1 above)
Comment (satta) [reply tests/ja4-quic/test.yaml:2]: True.

File: tests/ja4-rules-requires-off/test.yaml (hunk 3 above)
Comment (satta) [reply tests/ja4-rules-requires-off/test.yaml:6]: Yeah, right. You wouldn't even have "JA4 support" in the  `--build-info` output if there was no JA4 support in the code, even if is not enabled.

File: tests/tls/tls-ja3s-requires-off/test.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+pcap: ../../tls/tls-certs-alert/input.pcap
+
+requires:
+  min-version: 7.0.3
+  features:
+    - HAVE_LIBJANSSON
+    - HAVE_NSS
```
[the remark this answers is not in the document]
Comment (satta) [reply tests/tls/tls-ja3s-requires-off/test.yaml:7]: Totally. Will remove :)

File: tests/ja4-rules-requires/test.yaml (hunk 5 above)
Comment (satta) [reply tests/ja4-rules-requires/test.yaml:6]: In this case we want to test if the `requires` functionality works when JA4 support has been enabled at compile time.

File: tests/tls/tls-ja3s/test.yaml (hunk 6 above)
Comment (satta) [reply tests/tls/tls-ja3s/test.yaml:9]: I think I'll separate all JA3 related changes out into a separate commit; might be cleaner.

**Comment by satta** [comment]
Next: #1682

File: tests/ja4-rules-requires/test.yaml (hunk 5 above)
Comment (catenacyber) [reply tests/ja4-rules-requires/test.yaml:6]: This looks fragile, like if I rename `ProvidesFeature` to `ProvidingFeature`, this test will get skipped...

File: tests/ja4-rules-requires-off/test.yaml (hunk 4 above)
Comment (catenacyber) [reply tests/ja4-rules-requires-off/test.yaml:7]: ok, thanks, deserves a comment ;-)

**Pull request closed without merging**